sno/rich_tree_builder.py

```diff
@@ -13,18 +16,27 @@ class RichTreeBuilder:
     Conflicts are not detected.
     """
 
-    def __init__(self, repo, initial_root_tree):
+    def __init__(self, repo, initial_root_tree, auto_include_version_blobs=True):
         """
         The repo and an initial root tree which will be updated.
         All paths are specified relative to this tree - the root tree at a particular commit is a good choice.
```

Comment:
document what `auto_include_version_blobs` does, since it's not obvious?

Reply:
Actually, this should be at a different level - the tree builder should just build the tree its told to build and not be clever about it.
I've moved it to the caller - RepositoryStructure.create_tree_from_diff - and added a comment